graph/telemetry/istio/appender/kiali_annotation.go

```diff
@@ -0,0 +1,69 @@
+package appender
+
+import (
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/graph"
+	"github.com/kiali/kiali/models/annotation"
+)
+
+const AnnotationAppenderName = "kialiAnnotation"
+
+// Name: annotation
+type AnnotationAppender struct{}
+
+// Name implements Appender
+func (a AnnotationAppender) Name() string {
+	return AnnotationAppenderName
+}
+
+// AppendGraph implements Appender
+func (a AnnotationAppender) AppendGraph(trafficMap graph.TrafficMap, globalInfo *graph.AppenderGlobalInfo, namespaceInfo *graph.AppenderNamespaceInfo) {
+	if len(trafficMap) == 0 {
+		return
+	}
+
+	if getWorkloadList(namespaceInfo) == nil {
+		workloadList, err := globalInfo.Business.Workload.GetWorkloadList(namespaceInfo.Namespace)
+		graph.CheckError(err)
+		namespaceInfo.Vendor[workloadListKey] = &workloadList
+	}
+
```

Comment:
You don't want to get the ServiceDefinitionList every time, instead you want to grab the cached version that presumably would have been established by `IstioAppender`.  So insert here:

```
	if getServiceDefinitionList(namespaceInfo) == nil {
		sdl, err := globalInfo.Business.Svc.GetServiceDefinitionList(namespaceInfo.Namespace)
		graph.CheckError(err)
		namespaceInfo.Vendor[serviceDefinitionListKey] = sdl
	}
```